interface/batchcom/batchcom.php

```diff
@@ -24,280 +29,217 @@
-    if (!check_date_format($_POST['not_seen_since'])) $form_err.=xl('Date format for "not seen since" is not valid','','<br>');
+    if (!check_date_format($_POST['seen_before'])) $form_err.=xl('Date format for "seen before" is not valid','','<br>');
     // validate numbers
     if (!check_age($_POST['age_from'])) $form_err.=xl('Age format for "age from" is not valid','','<br>');
     if (!check_age($_POST['age_upto'])) $form_err.=xl('Age format for "age up to" is not valid','','<br>');
     // validate selections
-    if (!check_select($_POST['gender'],$gender)) $form_err.=xl('Error in "Gender" selection','','<br>');
-    if (!check_select($_POST['process_type'],$choices)) $form_err.=xl('Error in "Process" selection','','<br>');
-    if (!check_select($_POST['hipaa_choice'],$hipaa)) $form_err.=xl('Error in "HIPAA" selection','','<br>');
-    if (!check_select($_POST['sort_by'],$sort_by)) $form_err.=xl('Error in "Sort By" selection','','<br>');
+    if (!check_select($_POST['gender'],$gender_choices)) $form_err.=xl('Error in "Gender" selection','','<br>');
+    if (!check_select($_POST['process_type'],$process_choices)) $form_err.=xl('Error in "Process" selection','','<br>');
+    if (!check_select($_POST['hipaa_choice'],$hipaa_choices)) $form_err.=xl('Error in "HIPAA" selection','','<br>');
+    if (!check_select($_POST['sort_by'],$sort_by_choices)) $form_err.=xl('Error in "Sort By" selection','','<br>');
     // validates and or
-    if (!check_yes_no ($_POST['and_or_gender'])) $form_err.=xl('Error in YES or NO option','','<br>');
-    if (!check_yes_no ($_POST['and_or_app_within'])) $form_err.=xl('Error in YES or NO option','','<br>');
-    if (!check_yes_no ($_POST['and_or_seen_since'])) $form_err.=xl('Error in YES or NO option','','<br>');
-    if (!check_yes_no ($_POST['and_or_not_seen_since'])) $form_err.=xl('Error in YES or NO option','','<br>');
+    if (!check_and_or ($_POST['and_or_gender'])) $form_err.=xl('Error in and/or option','','<br>');
+    if (!check_and_or ($_POST['and_or_app_within'])) $form_err.=xl('Error in and/or option','','<br>');
+    if (!check_and_or ($_POST['and_or_seen_within'])) $form_err.=xl('Error in and/or option','','<br>');
 
     //process sql
     if (!$form_err) {
-
-
-         $sql="select patient_data.*, cal_events.pc_eventDate as next_appt,cal_events.pc_startTime as appt_start_time,cal_date.last_appt,forms.last_visit from patient_data left outer join openemr_postcalendar_events as cal_events on patient_data.pid=cal_events.pc_pid and curdate() < cal_events.pc_eventDate left outer join (select pc_pid,max(pc_eventDate) as last_appt from openemr_postcalendar_events where curdate() >= pc_eventDate group by pc_pid ) as cal_date on cal_date.pc_pid=patient_data.pid left outer join (select pid,max(date) as last_visit from forms where curdate() >= date group by pid) as forms on forms.pid=patient_data.pid";
```

Comment:
The strategy to deal with sql injection is to:
1. here, create a $sqlBindingArray = array
2. Open up all the if below to standard psr where format where the conditional is on separate line. then if pass the if and add sql to sql string. then change the variable to binding with ? and then add the variable to the sqlBindingArray.
3. Then when run the sql command use: $res = sqlStatement($sql, $sqlBindingArray);
4. See below on how to deal with the ORDER BY